adapters/adquery/adquery.go

```diff
@@ -0,0 +1,211 @@
+)
+
+const (
+	defaultCurrency string = "PLN"
+	bidderName      string = "adquery"
+	prebidVersion   string = "server"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Adquery adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) buildRequest(bidReq *openrtb2.BidRequest, imp *openrtb2.Imp, ext *openrtb_ext.ImpExtAdQuery) *BidderRequest {
+	userId := ""
+	if bidReq.User != nil {
+		userId = bidReq.User.ID
+	}
+
+	return &BidderRequest{
+		V:                   prebidVersion,
+		PlacementCode:       ext.PlacementID,
+		AuctionId:           "",
+		BidType:             ext.Type,
+		AdUnitCode:          imp.TagID,
+		BidQid:              userId,
+		BidId:               fmt.Sprintf("%s%s", bidReq.ID, imp.ID),
+		Bidder:              bidderName,
+		BidderRequestId:     bidReq.ID,
+		BidRequestsCount:    1,
+		BidderRequestsCount: 1,
+		Sizes:               getImpSizes(imp),
+	}
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, nil
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	var result []*adapters.RequestData
+	var errs []error
+	for _, imp := range request.Imp {
+		ext, err := parseExt(imp.Ext)
+		if err != nil {
+			errs = append(errs, &errortypes.BadInput{err.Error()})
+			continue
+		}
+
+		requestJSON, err := json.Marshal(a.buildRequest(request, &imp, ext))
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		data := &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpoint,
+			Body:    requestJSON,
+			Headers: headers,
+		}
+		result = append(result, data)
+	}
+
+	return result, errs
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	respData, price, width, height, errs := parseResponseJson(responseData.Body)
+	if errs != nil && len(errs) > 0 {
+		return nil, errs
+	}
```

Comment:
`errs != nil` check can be removed.  `len(errs) > 0` handles both nil and not nil check.  Refer https://go.dev/play/p/S1qu_wv6Gfy